### PR TITLE
feat(spoolman): allow selecting table columns in spool selection dialog

### LINKED ISSUE #1336: Spoolman: Select fields shown in Fluidd
### Is your feature request related to a problem? Please describe

Is there any way to select the options from Spoolman that are shown in Fluidd?
eg i do not need the fields like charge, comment or location shown in Fluidd.
But I do not find the option to select the fields that are shown.
Anybody who can help me?
Greetings

### Describe the solution you'd like

no solution

### Describe alternatives you've considered

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by matmen**
Closes #1336

image

### CODE REVIEW

File: src/store/config/state.ts [hunk 1]
```diff
@@ -116,6 +116,14 @@ export const defaultState = (): ConfigState => {
...
           { value: 'time_added', visible: true },
           { value: 'time_in_queue', visible: false }
+        ],
+        spoolman: [
+          { value: 'filament_name', visible: true },
+          { value: 'id', visible: true },
+          { value: 'material', visible: true },
+          { value: 'location', visible: true },
+          { value: 'comment', visible: true },
+          { value: 'last_used', visible: true }
```
Comment (matmen) [inline src/store/config/state.ts:126]: Not sure these are necessary, I think they're just for hidden-by-default columns?

Comment (pedrolamas) [reply src/store/config/state.ts:126]: Indeed, all we need here is the hidden columns, so I would revert the changes on this file.

Comment (pedrolamas) [reply src/store/config/state.ts:126]: I stand corrected: the entry IS required, but only needs the columns with `visible:false`, the rest is not required.

Given this, using `[]` should be ok!

**Comment by matmen** [comment]
>... but honestly I would leave it as you currently have it on this PR.

Oops, only saw that now. If either way works I'd prefer the empty array tho, that way it's less changes across files when more columns are added.

**Comment by matmen** [comment]
Oh weird, I didn't get a notification for that. I'll check it out right now!

**Review by pedrolamas — APPROVED** [review]

**Pull request merged**